azure-iot-provisioning-devicessdk/samples/register_symmetric_key.py

```diff
@@ -0,0 +1,54 @@
+import os
+import logging
+from azure.iot.provisioning.devicessdk.security.sk_security_provider import (
+    SymmetricKeySecurityProvider,
+)
+from azure.iot.provisioning.devicessdk.device.registration_client_factory import (
+    create_from_security_provider,
+)
+from azure.iot.hub.devicesdk import DeviceClient
+from azure.iot.hub.devicesdk.auth.authentication_provider_factory import from_connection_string
+
+import time
+
+logging.basicConfig(level=logging.INFO)
+
+provisioning_host = os.getenv("PROVISIONING_HOST")
+id_scope = os.getenv("PROVISIONING_IDSCOPE")
+registration_id = os.getenv("PROVISIONING_REGISTRATION_ID")
+symmetric_key = os.getenv("PROVISIONING_SYMMETRIC_KEY")
+
+
+symmetric_key_security_provider = SymmetricKeySecurityProvider(
+    registration_id, symmetric_key, id_scope
+)
+provisioning_client = create_from_security_provider(
+    provisioning_host, symmetric_key_security_provider, "mqtt"
+)
+
+
+def registration_status_callback(topic, payload):
+    result = str(payload)
+    print(result)
+    if "operationId" in result:
+        print("Device is registering")
+
+
+provisioning_client.on_registration_complete = registration_status_callback
```

Comment:
>registration_status_callback [](start = 47, length = 28)

Given what this does, it's kind of difficult to figure out another way to do this, but I really, really, really hate callbacks, and we have to at some point figure out a way to replace this pattern.

Reply:
wont fix right now.

---
In reply to: [265322440](https://github.com/Azure/azure-iot-sdk-python-preview/pull/59#discussion_r265322440) [](ancestors = 265322440)